go-controller/pkg/clustermanager/default_network_cluster_controller.go

```diff
@@ -0,0 +1,400 @@
+package clustermanager
+
+import (
+	"fmt"
+	"net"
+	"reflect"
+	"sync"
+
+	corev1 "k8s.io/api/core/v1"
+	cache "k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+
+	hotypes "github.com/ovn-org/ovn-kubernetes/go-controller/hybrid-overlay/pkg/types"
+	houtil "github.com/ovn-org/ovn-kubernetes/go-controller/hybrid-overlay/pkg/util"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/clustermanager/subnetallocator"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/kube"
+	objretry "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/retry"
+	ovntypes "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+)
+
+type networkClusterControllerBase struct {
+	kube         kube.Interface
+	watchFactory *factory.WatchFactory
+	stopChan     chan struct{}
+	wg           *sync.WaitGroup
+
+	// node events factory handler
+	nodeHandler *factory.Handler
+
+	// retry framework for nodes
+	retryNodes *objretry.RetryFramework
+}
+
+// defaultNetworkClusterController is the cluster controller for the default network
+// for all the nodes. Listens to the node events and allocates subnet from the
+// default cluster subnet pool.
+type defaultNetworkClusterController struct {
+	networkClusterControllerBase
+	defaultNetworkController *NodeNetworkController
```

Comment:
Nesting controllers would be a really hot topic. But I don't think NodeNetworkController is really a controller so I think we need to find a better name for it.